lib/spack/docs/packaging_guide.rst

```diff
@@ -2071,19 +2150,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
I believe there is inconsistency on which methods get `spec` and `prefix`, and which do not.  For example, I believe that `configure_args()` does not?  This should be pointed out.  Or even better... we agree to get rid of these convenience args across the board.  I think they are of diminishing convenience, since we write fewer `install()` methods.